Draft-Accepted/RFCXXXX-Command-Shell-Integration.md

```diff
@@ -0,0 +1,287 @@
+    },
+    "tags": {
+      "RequestID": "1111"
+    },
+    "type": "Microsoft.Resources/resourceGroups"
+  },
+  {
+    "id": "/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/RG2",
+    "location": "westcentralus",
+    "managedBy": null,
+    "name": "RG2",
+    "properties": {
+      "provisioningState": "Succeeded"
+    },
+    "tags": {
+      "RequestID": "1111"
+    },
+    "type": "Microsoft.Resources/resourceGroups"
+  }
+]
+```
+
+Although in PowerShell, this output can be piped to `ConvertFrom-Json`, it's an additional step for every `az` command execution.
+It would be preferrable to have the tool output structured data in a format that PowerShell can automatically convert to an object
+and used similarly to a cmdlet.
+
+Proposed output with custom formatting showing the most relevant information as well as a nested property:
+
+```output
+id                                                                                       location      name                  provisioningState
+--                                                                                       --------      ----                  ----------
+/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/ERNetwork-PvtApp      westcentralus RG1                   Succeeded
+/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/ERNetwork-SVC         westcentralus RG2                   Succeeded
+```
+
+### Tab Completion
+
+```powershell
+az g<tab>
+```
+
+This should result in:
+
+```powershell
+az group
+```
+
+As the tab completer would automtically be discovered by PowerShell and used.  In this example, there is only one subcommand starting with `g`.
+
+### Help
+
+```powershell
+get-help az
+```
+
+The help content would preferrably be in markdown with advanced rendering available in the shell.
+Normal help searches would also look within the help content defined by the tool.
+
+### Predictors and Feedback Providers
+
+PowerShell has an extensibility model for Predictors and Feedback Providers.
+Typically, these are implemented as modules with a cmdlet to register them with the PowerShell engine.
+In this case, the extension would still need to be managed code, but would be automatically discovered and loaded by the shell.
+Since the user installed the tool, the expectation is they want to use it.
+
+Based on user feedback, there can be a configuration file to disable specific extensions.
+
+## Specification
+
+A JSON manifest file would be placed alongside the executable (or technically anywhere within `$env:PATH`) and discovered by the shell.
+All paths are relative to the location of the manifest file and must use the forward slash path separator.
+
+### Command shell integration manifest
+
+The JSON manifest file name would be `<name>.command.json`.
+The `<name>` portion is only used to differentiate between multiple manifests and does not need to match the executable name,
+but it is recommended to do so.
```

Comment:
Your proposal seems reasonable.  The key thing is that at runtime, the executable is from the manifest content and not the filename of the manifest.  I don't know how often executable names collide, but it seems like it would be rare.  A convention like: `<executable>.<org/author>.command.json` seems fine.

Reply:
Adding this to update